decision records/003-Depenedency-Manager.md

```diff
@@ -0,0 +1,25 @@
+# Dependency Manager 
+•	Status: Proposed
+•	Issue: #638
+
+## Context and Problem Statement 
+As the number of packages increases in large scale projects, with multiple dependencies, updating package versions and dependencies manually is time consuming and takes great care to update dependencies one by one. 
+
+Other solutions encountered were local bash scripts as documented in both the issue https://github.com/Accenture/sfpowerscripts/issues/638 raised and in the discussions has here: https://github.com/Accenture/sfpowerscripts/discussions/468 
+
+## Solution
+The dependency manager should utilise the SemVer incrementation style https://semver.org/ and be of a similar method to the ‘lerna version’ command. It should prompt the user for the version: (major, minor, patch, custom). 
+Since salesforce does not necissarily follow the semver versioning, we will provide a summary to the bump. 
+Eg. Option 1 bump Major - includes incompatible API changes
+We will also not be supporting pre-release versioning at this stage. 
```

Comment:
Can you elaborate here? What does pre-release mean?

Reply:
A pre-release version MAY be denoted by appending a dash and a series of dot separated identifiers immediately following the patch version. Identifiers MUST be comprised of only ASCII alphanumerics and dash [0-9A-Za-z-]. Pre-release versions satisfy but have a lower precedence than the associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92.

Reply:
yeah we dont need pre-release support, as salesforce packages dont support it. We need to find a way for build number for source and data packages, say if the user has any non zero build number, do we bring back to zero?

Reply:
We should check if the number is non-zero and ask if they want it reset to zero if it is

Reply:
can you add these to the document?